python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,371 @@
+    """Config for star galaxy classifier
+    """
+    version = pexConfig.Field(
+        doc="Version information",
+        dtype=str,
+        default="HSC-COSMOS-20180528",
+    )
+
+    pass
+
+
+class StarGalaxyClassifierTask(Task):
+    """Add classification (star or galaxy) and associated flag columns to a parquet table.
+    """
+
+    _DefaultName = "starGalaxyClassifier"
+    ConfigClass = StarGalaxyClassifierConfig
+
+    def __init__(self):
+        Task.__init__(self)
+
+    def runDataRef(self, dataRef):
+        """Add required magnitudes and run the classifier
+
+        checks that all the required filters are in the file, then adds magntiudes and runs the classifier
+        for the given patch and tract.
+        """
+        cat = dataRef.get("deepCoadd_obj").toDataFrame({"dataset": "meas"})
+
+        # For now raise a not implemented error for other cameras, eventually needs other trained classifiers
+        camera_name = dataRef.getButler()._getDefaultMapper().getCameraName()
+        if camera_name != "hsc":
+            raise NotImplementedError
+
+        # To Do: DM-14539 - Move this to somewhere else
+        # put this into /datasets/hsc then the butler will get it
+        # Add classifier_pickle to obs_base datasets, copy deepCoadd_skyMap.
+        clfFilenameMorph = "/home/sr525/starGalaxy/clfIsotonicMorph.pickle"
+        with open(clfFilenameMorph, "rb") as f:
+            filters = pickle.load(f)
+
+        filtersInCat = set(cat.columns.get_level_values(0))
+        if filtersInCat != set(filters):
+            raise RuntimeError("Not all filters required in file")
+
+        # addMagnitdues will probably be removed later as it will hopefully be superseded by Tim
+        cat = addMagnitudes(cat, dataRef, filters)
+        cat = self.run(cat, filters)
+
+        cat = ParquetTable(dataFrame=cat)
+        dataRef.put(cat, "deepCoadd_sg")
+
+    def run(self, cat, filters):
+        """Add colour based classification, morphology only based classification and their associated flags.
+
+        Parameters
+        ----------
+        cat : 'Pandas data frame'
+        filters : 'list'
+            List of the filters for the camera used
+
+        Returns
+        -------
+        Data frame with the two classification and two flag columns added and a new index containing the
+        columns required by the classifier
+
+        Notes
+        -----
+        For the morphology only classifier the features used are psf - cmodel in r, i and z, m, b, seeing in
+        g, r, i, z and Y, flux/flux error in g, r, i, z and Y, psf - cmodel error in r, i and z
+        For the colour and morphology classifier the features used are g - r, r - i, i - z, z - Y (all using
+        cmodel), psf - cmodel in g, r, i, z and Y, m, b, flux / flux error in g, r, i, z and Y, seeing in g,
+        r, i, z and Y, psf - cmodel error in g, r, i, z and Y
+        The columns must be used in this order
```

Comment:
This is very helpful, but I am puzzled as to why you have explicit filters listed here yet `filters` is an argument. How does `filters` relate to the filters listed above?

If the filters are truly hard-coded that is very worrying. They should be an argument or a config parameter if at all possible.

What do you mean by "the columns must be used in this order"? Please say more about this in the doc string.

How do you decide whether to use a morphology-only classifier or a colour and morphology classifier? I would expect this to be a config parameter (e.g. a flag or  subtask that can be retarged) or an argument (e.g. a flag or enum).

Reply:
The filter names were left in from before I generalised the code, sorry. The code currently applies both classifiers as they each perform better in different situations.